### PR TITLE
feat(cmd): avoid to use CommandAttributes::flags directly

### PR DESCRIPTION
**Pull request by PragmaTwice**
We should use `CommandAttributes::GenerateFlags` instead of accessing the `flags` member directly.

### CODE REVIEW

**Comment by mapleFU** [comment]
Is this to avoid not calling `flag_gen` fn if having?

**Comment by PragmaTwice** [comment]
> Is this to avoid not calling `flag_gen` fn if having?

Exactly.

**Comment by mapleFU** [comment]
What about making it private to avoid some explicit calling? ( Current method lgtm but I think it's easy to use it directly next time)

**Comment by PragmaTwice** [comment]
> What about making it private to avoid some explicit calling? ( Current method lgtm but I think it's easy to use it directly next time)

Done.

**Comment by PragmaTwice** [comment]
Also I plan to make key_range and other related members private in the next PR.

File: src/commands/commander.h [hunk 1]
```diff
@@ -182,9 +228,11 @@ struct CommandAttributes {
   // commander object generator
   CommanderFactory factory;
 
+  uint64_t InitialFlags() const { return flags_; }
```
Comment (mapleFU) [inline src/commands/commander.h:231]: Minor question: why `InitialFlags()` in some cases?

Comment (PragmaTwice) [reply src/commands/commander.h:231]: Some times we cannot get the cmd args, so maybe the initial flags is useful. Also I currently don't want to touch some legacy sources.

File: src/commands/commander.h [hunk 2]
```diff
@@ -257,23 +312,21 @@ inline uint64_t ParseCommandFlags(const std::string &description, const std::str
   return flags;
 }
 
+template <typename T>
+auto MakeCmdAttr(const std::string &name, int arity, const std::string &description, NoKeyInThisCommand no_key,
+                 const AdditionalFlagGen &flag_gen = {}) {
+  CommandAttributes attr(name, arity, CommandCategory::Unknown, ParseCommandFlags(description, name), flag_gen, no_key,
```
Comment (mapleFU) [inline src/commands/commander.h:318]: So NO_KEY has becoming a different ctor fn?

**Review by mapleFU — APPROVED** [review]

File: src/commands/commander.h (hunk 2 above)
Comment (PragmaTwice) [reply src/commands/commander.h:318]: Yeah, `0,0,0` is not clear enough.

**Pull request merged**